Add fail-on-error: false for Coveralls CI step

### Description

The Coveralls service is used in this repository's CI pipeline to publish and report on test coverage results. However, Coveralls has on multiple occasions experienced service interruptions or internal server errors (e.g., HTTP 500). 
See e.g.: 

- [https://github.com/lemurheavy/coveralls-public/issues/1816](https://github.com/lemurheavy/coveralls-public/issues/1816)
- [https://github.com/lemurheavy/coveralls-public/issues/1708](https://github.com/lemurheavy/coveralls-public/issues/1708)

These issues disrupt the ubuntu.yml workflow entirely and fails it, even if nothing changed within the nlohmann/json repo. 

Following the most recent incident on 14/8/2025, a postmortem published by Coveralls recommends as best practice to include the input option "fail-on-error: false" in the configuration of Coveralls (see [https://status.coveralls.io/incidents/v5mcbrsbhgt4](https://status.coveralls.io/incidents/v5mcbrsbhgt4)). 

Note: the postmortem includes a typo, the parameter is "fail-on-error" and not "continue-on-error", as can be seen  in the list of inputs here: https://github.com/marketplace/actions/coveralls-github-action

The proposed improvement ensures that the workflow does not terminate or fail if Coveralls encounters temporary issues, such as API returning HTTP 500 errors or outages. 

### Reproduction steps

Running the workflow "ubuntu.yml" when Coveralls is experiencing an internal server error.

### Expected vs. actual results

Expected: Proceed with other steps in ubuntu.yml workflow even if Coveralls encounters service errors.


Actual results: Even with no changes made to the repo, the workflow failed, due to Coveralls servers being down. Although the LCOV coverage report was still created and archived.

### Minimal code example

```Shell
- name: Publish report to Coveralls
  uses: coverallsapp/github-action@648a8eb78e6d50909eff900e4ec85cab4524a45b # v2.3.6
  with:
    github-token: ${{ secrets.GITHUB_TOKEN }}
    path-to-lcov: ${{ github.workspace }}/build/json.info.filtered.noexcept
```

### Error messages

```Shell
🚀 Posting coverage data to https://coveralls.io/api/v1/jobs
⚠️ Internal server error. Please contact Coveralls team.
Error: Process completed with exit code 1.
```

### Compiler and operating system

Windows, GCC-11

### Library version

3.12.0

### Validation

- [x] The bug also occurs if the latest version from the [`develop`](https://github.com/nlohmann/json/tree/develop) branch is used.
- [x] I can successfully [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests).

--- a/.github/workflows/ubuntu.yml
+++ b/.github/workflows/ubuntu.yml
@@ -124,6 +124,7 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           path-to-lcov: ${{ github.workspace }}/build/json.info.filtered.noexcept
+          fail-on-error: false
 
   ci_test_compilers_gcc_old:
     runs-on: ubuntu-latest

